### DISCUSSION TITLE
How to use react-aria with jest snapshot testing?

### DISCUSSION THREAD

**Discussion by chocnut** [adobe/react-spectrum#4685] [Q&A] [answered]
Hello, I'm currently incorporating react-aria into our project and using jest for snapshot testing. Unfortunately, the snapshot failed due to issues with dynamic aria properties. I was wondering if anyone could provide me with some guidance on this matter. Your assistance would be greatly appreciated. Thank you.

**Comment by reidbarber** [comment, marked as answer]
Suggested by @LFDanLu in https://github.com/adobe/react-spectrum/issues/2272#issuecomment-911897674:

> Does wrapping your test components in a SSRProvider help? Otherwise maybe you could try [mocking useId](https://github.com/adobe/react-spectrum/issues/1167#issuecomment-899753521) as a work around

**Reply by chocnut** [reply]
Thank you very much for answering my question. I will try it out.
